ui/admin/app/components/form/field/list-wrapper/key-value/index.js

```diff
@@ -29,6 +29,20 @@ export default class MappingListComponent extends Component {
   get options() {
     return this.args?.options || this.args?.model?.[this.args.name];
   }
+
+  /**
+   * Determines if we need to show an empty row to the users to enter more key/value pairs based on removeDuplicates arg,
+   * by default it is true
+   * @type {object}
+   */
+  get showNewRow() {
+    if (this.args.removeDuplicates) {
+      return this.options?.length !== this.args.allowedEntries;
```

Comment:
Shouldn't we pass in the entire function `showNewRow` and let the user handle this logic rather than passing `removeDuplicates` and `allowedEntries`? We can check if `this.args.showNewRow` is a func and then call it if it is.

Reply:
good call, I moved it to consumer and passed the func to component